src/bin/pg_autoctl/pghba.c

```diff
@@ -172,6 +172,59 @@ pghba_ensure_host_rule_exists(const char *hbaFilePath,
 }
 
 
+/*
+ * write_trust_local_hba_rule gets the hbaFilePath and overrides it with
+ * 		"local all all  trust"
+ * 	which means allow all connections via unix domain sockets.
+ */
+bool
+write_trust_local_hba_rule(const char *hbaFilePath)
+{
+	const char *userName = "all";
+	const char *authenticationScheme = "trust";
+
+	PQExpBuffer hbaLineBuffer = createPQExpBuffer();
+	if (hbaLineBuffer == NULL)
+	{
+		log_error("Failed to allocate memory");
+		return false;
+	}
+
+	appendPQExpBufferStr(hbaLineBuffer, "local ");
+
+	append_database_field(hbaLineBuffer, HBA_DATABASE_ALL, NULL);
+	appendPQExpBufferStr(hbaLineBuffer, " ");
+
+	appendPQExpBufferStr(hbaLineBuffer, userName);
+	appendPQExpBufferStr(hbaLineBuffer, " ");
+
+	appendPQExpBuffer(hbaLineBuffer, " %s", authenticationScheme);
+
+	appendPQExpBufferStr(hbaLineBuffer, HBA_LINE_COMMENT "\n");
+
+	if (PQExpBufferBroken(hbaLineBuffer))
+	{
+		log_error("Failed to allocate memory");
+		destroyPQExpBuffer(hbaLineBuffer);
+		return false;
+	}
+
+	log_debug("generated monitor pg_hba rule for unix domain socket: \"%s\"", hbaLineBuffer->data);
+
+	/* write the resulting content at the destination path */
+	if (!write_file(hbaLineBuffer->data, hbaLineBuffer->len, hbaFilePath))
+	{
+		/* error is already logged */
+		destroyPQExpBuffer(hbaLineBuffer);
+		return false;
+	}
```

Comment:
What would be good here would be to follow Jelte's remark from the other PR and allow us to keep the useful comments from the main pg_hba.conf file. I just had a look at it seems to me that there's a rather easy way to do that: as long as the lines start with a `#` character, keep it. First time it's not the case, stop copying from the existing HBA file, add an empty line, then our own stuff.

If we do that we lose the following table-like header, but we could always manually re-add it before we add our own lines, and then we could use `%25s` or whatever to respect the table like formatting of the HBA rules:

```
# TYPE  DATABASE        USER            ADDRESS                 METHOD
```

Then the default HBA file could be kept under the name `pg_hba.default.conf` or something.

Reply:
See the `splitLines` function in `string_utils.c` and its call sites for an example how to easily achieve that.